### PR TITLE
Add torch.cat tensors type promotion description

### LINKED ISSUE #126964: Implicit data type promotion in torch.cat is undocumented
### 📚 The doc issue

In the [torch.cat doc](https://pytorch.org/docs/stable/generated/torch.cat.html) it says

```
Parameters

- tensors (sequence of Tensors) – any python sequence of tensors of the same type. 

```

But in reality mixing tensors with different data types works well on both cpu and cuda.

```
>>> import torch
>>> x = torch.randn(2, 3,dtype=torch.float32,device="cuda")
>>> x
tensor([[ 1.8556,  2.3431,  0.3822],
        [-0.2561,  0.7483,  0.3750]], device='cuda:0')
>>> y = torch.randn(2, 3,dtype=torch.float64,device="cuda")
>>> y
tensor([[ 1.4577,  1.6310,  0.6145],
        [ 1.2279, -0.2790,  1.0286]], device='cuda:0', dtype=torch.float64)
>>> z=torch.cat((x,y),0)
>>> z
tensor([[ 1.8556,  2.3431,  0.3822],
        [-0.2561,  0.7483,  0.3750],
        [ 1.4577,  1.6310,  0.6145],
        [ 1.2279, -0.2790,  1.0286]], device='cuda:0', dtype=torch.float64)

```

It seems the implicit dtype promotion is handled by the structured kernel boilerplate code?

### Suggest a potential alternative/fix

I don't know whether this is a bug or feature. Semantically user shouldn't concat tensors with different dtypes.

If it's a bug, should we put some parameter sanitization code in torch.cat that prevent mixing dtypes.

Or if it's a feature, do we need to change the doc about the `tensors of the same type` part.

cc @svekars @brycebortree

### PR DESCRIPTION
**Pull request by zeshengzong**
Fixes #126964

Add note description about type promotion of `torch.cat`

**Test Result**

**Before**
image

**After**
image

cc @albanD @svekars

### CODE REVIEW

**Comment by zeshengzong** [comment]
@albanD please review the change when available, thanks!

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -2293,14 +2293,18 @@ def merge_dicts(*dicts):
...
+        except in the cat dimension.
+
     dim (int, optional): the dimension over which the tensors are concatenated
 
 Keyword args:
     {out}
 
+.. note:: It is recommended to keep all `tensors` in the same dtype to avoid unintended type promotion.
+          For example, mixing `torch.int32` and `torch.int64` may result in the output being automatically
+          promoted to `torch.int64`.
```
Comment (albanD) [inline torch/_torch_docs.py:2306]: I'm not sure this is needed. All functions in PyTorch have this behavior of doing type promotion.
If you think it is necessary to leave a comment here, you can make it one sentence and make it not a "may" but "will"

Comment (zeshengzong) [reply torch/_torch_docs.py]: If it's a common sense, I think unnecessary to declare this behavior in the method, removed!
Thanks!

**Review by albanD — APPROVED** [review]
Thanks for fixing the wrong phrasing!

**Comment by albanD** [comment]
@pytorchbot merge

I had to trigger the CI again before the merge could happen!

**Pull request closed without merging**